### PR TITLE
Enable acmp driver for imxrt

### PR DESCRIPTION
**Pull request by ChayGuo**
Updated mcu_acmp driver to support discrete mode configuration for i.MXRT11xx.
Add ACMP support on MIMXRT1170 EVK

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]
LGTM

**Comment by ChayGuo** [comment]
The documentation build failure seems not related with my patches. 
`warning: LV_Z_HOR_RES_MAX (defined at boards/arm/mimxrt10[50](https://github.com/zephyrproject-rtos/zephyr/runs/6446979569?check_suite_focus=true#step:7:50)_evk/Kconfig.defconfig:64, boards/arm/mimxrt1060_evk/Kconfig.defconfig:65, boards/arm/mimxrt1064_evk/Kconfig.defconfig:59) defined without a type
warning: LV_Z_VER_RES_MAX (defined at boards/arm/mimxrt1050_evk/Kconfig.defconfig:67, boards/arm/mimxrt1060_evk/Kconfig.defconfig:68, boards/arm/mimxrt1064_evk/Kconfig.defconfig:62) defined without a type
warning: LV_Z_DISPLAY_DEV_NAME (defined at boards/arm/stm32f429i_disc1/Kconfig.defconfig:17) defined without a type`

File: samples/sensor/mcux_acmp/README.rst [hunk 1]
```diff
@@ -26,3 +31,13 @@ ACMP input voltage by turning the on-board potentiometer.
    :board: twr_ke18f
    :goals: flash
    :compact:
+Building and Running for MIMXRT1170-EVK
```
Comment (danieldegrasse) [inline samples/sensor/mcux_acmp/README.rst:34]: The lack of blank line on line 33 is likely causing the CI error here:
` /home/runner/work/zephyr/zephyr/doc/_build/src/samples/sensor/mcux_acmp/README.rst:34:Explicit markup ends without a blank line; unexpected unindent.`

Comment (ChayGuo) [reply samples/sensor/mcux_acmp/README.rst:35]: > The lack of blank line on line 33 is likely causing the CI error here: ` /home/runner/work/zephyr/zephyr/doc/_build/src/samples/sensor/mcux_acmp/README.rst:34:Explicit markup ends without a blank line; unexpected unindent.`

Thanks for identifying the issue. Fixed the blank line.

File: samples/sensor/mcux_acmp/sample.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ sample:
   description: Demonstration of the NXP MCUX ACMP sensor driver
   name: NXP MCUX ACMP sample
 common:
-  platform_allow: twr_ke18f
+  platform_allow: twr_ke18f mimxrt1170_evk_cm7 mimxrt1170_evk_cm4
```
Comment (danieldegrasse) [inline samples/sensor/mcux_acmp/sample.yaml:5]: If CM4 is enabled for testing, should there be a board overlay for it as well?

File: include/zephyr/drivers/sensor/mcux_acmp.h [hunk 3]
```diff
@@ -50,6 +75,22 @@ enum sensor_attribute_mcux_acmp {
...
+	SENSOR_ATTR_MCUX_ACMP_POSITIVE_DISCRETE_MODE,
+	/** Analog Comparator Negative Channel Discrete Mode Enable. */
+	SENSOR_ATTR_MCUX_ACMP_NEGATIVE_DISCRETE_MODE,
+	/** Analog Comparator discrete mode clock selection. */
+	SENSOR_ATTR_MCUX_ACMP_DISCRETE_CLOCK,
+	/** Analog Comparator resistor divider enable. */
+	SENSOR_ATTR_MCUX_ACMP_DISCRETE_ENABLE_RESISTOR_DIVIDER,
+	/** Analog Comparator discrete sample selection. */
+	SENSOR_ATTR_MCUX_ACMP_DISCRETE_SAMPLE_TIME,
+    /** Analog Comparator discrete phase1 sampling time selection. */
```
Comment (gmarull) [inline include/zephyr/drivers/sensor/mcux_acmp.h:89]: nit: fix indentation here

File: include/zephyr/drivers/sensor/mcux_acmp.h [hunk 4]
```diff
@@ -18,6 +19,30 @@ extern "C" {
 
 #include <zephyr/drivers/sensor.h>
 
+#if defined(FSL_FEATURE_ACMP_HAS_C1_INPSEL_BIT) && (FSL_FEATURE_ACMP_HAS_C1_INPSEL_BIT == 1U)
+#define MCUX_ACMP_HAS_INPSEL 1
+#else
+#define MCUX_ACMP_HAS_INPSEL 0
+#endif
```
Comment (gmarull) [inline include/zephyr/drivers/sensor/mcux_acmp.h:26]: suggestion, here and others:

```c
#define MCUX_ACMP_HAS_INPSEL \
    (defined(FSL_FEATURE_ACMP_HAS_C1_INPSEL_BIT) && (FSL_FEATURE_ACMP_HAS_C1_INPSEL_BIT == 1U))
```

**Review by gmarull — DISMISSED** [review]
lgtm, minor comments

**Review by henrikbrixandersen — DISMISSED** [review]

File: samples/sensor/mcux_acmp/sample.yaml (hunk 2 above)
Comment (ChayGuo) [reply samples/sensor/mcux_acmp/sample.yaml:5]: > here be a board overlay fo

Yes， both cm4 and cm7 overlay added.

File: include/zephyr/drivers/sensor/mcux_acmp.h (hunk 3 above)
Comment (ChayGuo) [reply include/zephyr/drivers/sensor/mcux_acmp.h:89]: > nit: fix indentation here

Fixed. thanks.

File: include/zephyr/drivers/sensor/mcux_acmp.h (hunk 4 above)
Comment (ChayGuo) [reply include/zephyr/drivers/sensor/mcux_acmp.h:26]: > suggestion, here and others:
> 
> ```c
> #define MCUX_ACMP_HAS_INPSEL \
>     (defined(FSL_FEATURE_ACMP_HAS_C1_INPSEL_BIT) && (FSL_FEATURE_ACMP_HAS_C1_INPSEL_BIT == 1U))
> ```

Accepted and updated, thanks.

**Comment by ChayGuo** [comment]
Updated the PR per review comments.

**Review by danieldegrasse — APPROVED** [review]

File: samples/sensor/mcux_acmp/boards/mimxrt1170_evk_cm4.overlay [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/*
```
Comment (mmahadevan108) [inline samples/sensor/mcux_acmp/boards/mimxrt1170_evk_cm4.overlay:1]: can these configurations move to the boards folder?

File: samples/sensor/mcux_acmp/boards/mimxrt1170_evk_cm7.overlay [hunk 6]
```diff
@@ -0,0 +1,22 @@
+/*
```
Comment (mmahadevan108) [inline samples/sensor/mcux_acmp/boards/mimxrt1170_evk_cm7.overlay:1]: can these configurations move to the boards folder?

File: samples/sensor/mcux_acmp/src/main.c [hunk 7]
```diff
@@ -28,13 +34,17 @@ struct acmp_attr {
 };
 
 static const struct acmp_attr attrs[] = {
+#if MCUX_ACMP_HAS_INPSEL
```
Comment (mmahadevan108) [inline samples/sensor/mcux_acmp/src/main.c:37]: Is this change needed. If so, then a similar change should be made to the driver and include files

File: samples/sensor/mcux_acmp/src/main.c [hunk 8]
```diff
@@ -28,13 +34,17 @@ struct acmp_attr {
...
 	/* Positive input port set to MUX */
 	{ .attr = SENSOR_ATTR_MCUX_ACMP_POSITIVE_PORT_INPUT, .val = 1 },
+#endif
 	/* Positive input channel */
 	{ .attr = SENSOR_ATTR_MCUX_ACMP_POSITIVE_MUX_INPUT,
 	  .val = ACMP_POSITIVE },
+#if MCUX_ACMP_HAS_INNSEL
```
Comment (mmahadevan108) [inline samples/sensor/mcux_acmp/src/main.c:44]: Same here

File: samples/sensor/mcux_acmp/src/main.c [hunk 9]
```diff
@@ -45,8 +55,14 @@ static const struct acmp_attr attrs[] = {
 	{ .attr = SENSOR_ATTR_MCUX_ACMP_DAC_VALUE, .val = ACMP_DAC_VALUE },
 	/* Hysteresis level */
 	{ .attr = SENSOR_ATTR_MCUX_ACMP_HYSTERESIS_LEVEL, .val = 3 },
+#if MCUX_ACMP_HAS_DISCRETE_MODE
+	/* Discrete mode */
+	{ .attr = SENSOR_ATTR_MCUX_ACMP_POSITIVE_DISCRETE_MODE, .val = 1 },
+#endif
+#if MCUX_ACMP_HAS_OFFSET
```
Comment (mmahadevan108) [inline samples/sensor/mcux_acmp/src/main.c:62]: Same here.

File: samples/sensor/mcux_acmp/src/main.c (hunk 7 above)
Comment (ChayGuo) [reply samples/sensor/mcux_acmp/src/main.c:37]: If we set the unsupported attribute the mcux_acmp_attr_set will return -ENOTSUP error. So there are two potential resolutions:
1. Did not provide the attribute which not supported, this is what I did now, but need also made the change to the driver include files. 
2. Ignore the -ENOTSUP error in the application. 
I'm not sure which should be preferred.

File: samples/sensor/mcux_acmp/boards/mimxrt1170_evk_cm4.overlay (hunk 5 above)
Comment (ChayGuo) [reply samples/sensor/mcux_acmp/boards/mimxrt1170_evk_cm4.overlay:1]: It's kept aligned with twr_ke18f.overlay. Per my understanding, the acmp1 in2 is only used for the sample test, there may be pin conflicts with other modules and there's no obviously fixed map of comparator on the board.

File: samples/sensor/mcux_acmp/boards/mimxrt1170_evk_cm7.overlay (hunk 6 above)
Comment (ChayGuo) [reply samples/sensor/mcux_acmp/boards/mimxrt1170_evk_cm7.overlay:1]: Same as above.

**Review by mmahadevan108 — APPROVED** [review]
